test/ArrayField_test.js

```diff
@@ -38,21 +38,21 @@ describe("ArrayField", () => {
       expect(legend.id).eql("root__title");
     });
 
-    it("should contain no field in the list by default", () => {
+    it("should contain no field in the list by default", function*() {
```

Comment:
As stated in the description, I wish I could scientifically elaborate, but I can't. I *suspect* the trick is that the babel polyfill for generators uses `setTimeout` (or `setImmediate`) behind the hood and *maybe* that ensures us having at least a tick passed for each contained statement, which addresses our test use cases just fine.

Also, I just blindly migrated all `it` statements to leverage this *feature* to keep things consistent, and also because before this change we simply add 158 test failures (and zero after the migration).

If you're interested in investigating this, it's probably worth inspecting the babel polyfill code for generators, to see what they're doing precisely, and figure out how it actually helps us writing synchronous test code compatible with the asynchronous nature of our current expectations (which is inherently highly surprising anyway, to say the least).